### PR TITLE
Remove action link from Plugin Checker

### LINKED ISSUE #213: Remove `Check this plugin` action link from Plugin Checker
### Is your enhancement related to a problem? Please describe.

During my preparation for the internal demo, I discovered an issue with the "Check this plugin" action link in the Plugin Checker feature. Currently, this action link is available for the Plugin Checker, but it is unusable as we are unable to test the Plugin Checker functionality. To enhance the usability and overall user experience, I suggest removing the "Check this plugin" action link from the Plugin Checker.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
Closes #213 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by swissspidy — APPROVED** [review]

File: tests/phpunit/Admin/Admin_Page_Tests.php [hunk 1]
```diff
@@ -145,15 +145,7 @@ public function test_filter_plugin_action_links() {
...
-
-		$this->assertEquals(
-			sprintf(
-				'<a href="%1$s">%2$s</a>',
-				esc_url( admin_url( "tools.php?page=plugin-check&plugin={$base_file}" ) ),
-				esc_html__( 'Check this plugin', 'plugin-check' )
-			),
-			$action_links[0]
-		);
+		$this->assertEmpty( $action_links );
```
Comment (jjgrainger) [inline tests/phpunit/Admin/Admin_Page_Tests.php]: @mukeshpanchal27 is this test correct? I looks like we assert the same thing twice but also not testing for when the link is added?

Can we add a test/assertion for when the Check this plugin link is added?

**Review by jjgrainger — CHANGES_REQUESTED** [review]
Thanks @mukeshpanchal27 looks good but have a question on the updates to the tests here

File: tests/phpunit/Admin/Admin_Page_Tests.php (hunk 1 above)
Comment (spacedmonkey) [reply tests/phpunit/Admin/Admin_Page_Tests.php]: I think we should add another test for this?

Comment (felixarntz) [reply tests/phpunit/Admin/Admin_Page_Tests.php]: Agreed, this is currently a duplicate. We should have one test here for a regular plugin and ensure that the link is added, and then _another_ test for the plugin checker plugin and ensure that the link is not added.

**Review by felixarntz — COMMENTED** [review]
@mukeshpanchal27 Thanks for the PR, LGTM except for the test feedback.

File: tests/phpunit/Admin/Admin_Page_Tests.php (hunk 1 above)
Comment (mukeshpanchal27) [reply tests/phpunit/Admin/Admin_Page_Tests.php]: Thank you for the feedback. In https://github.com/10up/plugin-check/pull/225/commits/f490742b350daad20de612a30c7440774de2657f i have added new unit test as requested.

**Review by jjgrainger — APPROVED** [review]
Thanks @mukeshpanchal27 changes look good

**Review by felixarntz — APPROVED** [review]
Thanks @mukeshpanchal27, LGTM!

**Pull request merged**